cli/src/utils.ts

```diff
@@ -73,3 +80,54 @@ export const assertReplicaIsHealthy = async (network: string): Promise<void> =>
     throw new Error('The replica is not healthy.');
   }
 };
+
+export const commandExists = (command: string): boolean => {
+  const result = spawnSync('command', ['-v', command], { stdio: 'ignore' });
+  return result.status === 0;
+};
+
+export const assertCommandExists = (command: string): void => {
+  if (!commandExists(command)) {
+    throw new Error(`Command '${command}' does not exist.`);
+  }
+};
+
+export const getReplicaUrl = async (network: string): Promise<string> => {
+  const dfxFile = JSON.parse(await readFile(DFX_PATH, 'utf-8'));
+
+  if (!dfxFile?.networks?.[network]) {
+    throw new Error(`Network '${network}' not found in dfx.json.`);
+  }
+
+  if (dfxFile.networks[network].providers && dfxFile.networks[network].providers.length > 0) {
+    return dfxFile.networks[network].providers[0];
+  }
+
+  if (dfxFile.networks[network].bind) {
+    return dfxFile.networks[network].bind.startsWith('http')
+      ? dfxFile.networks[network].bind
+      : `http://${dfxFile.networks[network].bind}`;
```

Comment:
is this http:// a safe choice?

Reply:
Yes, bind is only used for localhost ifaik

Reply:
> bind is only used for localhost ifaik

how about failing the script if this assumption is wrong?